src/pgduckdb_duckdb.cpp

```diff
@@ -164,7 +164,11 @@ DuckDBManager::Initialize() {
 	auto &context = *connection->context;
 
 	auto &db_manager = duckdb::DatabaseManager::Get(context);
-	default_dbname = db_manager.GetDefaultDatabase(context);
+	if (duckdb_motherduck_default_database[0] == '\0') {
+		default_dbname = db_manager.GetDefaultDatabase(context);
+	} else {
+		default_dbname = duckdb_motherduck_default_database;
+	}
```

Comment:
Instead of overriding the default database that motherduck sets, let's instead make motherduck set the correct default database by providing the database in the connection string:

```c
connection_string = psprintf("md:%s?motherduck_token=%s", duckdb_motherduck_default_database, duckdb_motherduck_token);
```

For that you need to escape the database name though using URI escaping. I think the following function should work for that:
```c
char *
uri_escape(const char *input)
{
    StringInfoData buf;
    initStringInfo(&buf);

    while (*str)
    {
        char c = *str++;
        if ((c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9') || c == '-' || c == '_' || c == '.' || c == '~')
        {
            appendStringInfoChar(&buf, c);
        }
        else
        {
            appendStringInfo(&buf, "%%%02X", (unsigned char)c);
        }
    }

    return buf.data;
}
```

Reply:
Thanks for the tip! I didn't realize you could just stick the db name in the connection string, clean solution.

Is it ok that I just stuck the uri_escape function in the same file?